src/azure-cli/azure/cli/command_modules/acs/custom.py

```diff
@@ -2429,6 +2433,10 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
             cluster_identity_object_id,
             assign_kubelet_identity)
 
+    auto_upgrade_profile = None
```

Comment:
not necessary.
if someone write some code to set the value of auto_upgrade_profile above 2436.
then it will be set to None.

Reply:
The issue is that auto_upgrade_profile is used to pass into a function later on. If it is not set before that point from my understanding that will case an issue like follows:
![image](https://user-images.githubusercontent.com/33269602/130499728-578397a0-6a18-4f97-928d-dd067915e10b.png)

Reply:
Reference: https://www.pluralsight.com/guides/python-basics-variables-assignment

Reply:
Assign `auto_upgrade_profile = None` at the beginning of the function?

Reply:
It seems to me like the style in this function is to set the variable to None/default before whatever check there is to set it. Some examples:
identity_profile, identity, and api_server_access_profile directly before this.

Reply:
I vote for "Assign auto_upgrade_profile = None at the beginning of the function".